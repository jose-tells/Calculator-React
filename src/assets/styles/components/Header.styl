.section 
  display flex
  align-items center
  width 65%
  margin  38px 0 22px 13px
  .section__icon
    font-size 2rem
    margin-right 15px
    justify-self flex-end
    align-self center
    color button-blue
  .section__title
    align-self center
    font-weight 600
    color white
