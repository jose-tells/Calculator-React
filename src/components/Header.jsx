import React from 'react';
import { Link } from 'react-router-dom';
// Font awesome
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faChevronLeft } from '@fortawesome/free-solid-svg-icons';


const Header = (props) => {
  const { section } = props;

  return(
    <header className="section">
        <Link to="/">
          <FontAwesomeIcon className="section__icon" icon={faChevronLeft}/>
        </Link>
      <h1 className="section__title">{section}</h1>
    </header>
  )
};

export default Header;