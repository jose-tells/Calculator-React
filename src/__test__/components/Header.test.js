import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { mount } from 'enzyme';
import { create } from 'react-test-renderer';
import Header from '../../components/Header';

describe('<Header />', () => {
  const header = mount(
    <BrowserRouter>
      <Header />
    </BrowserRouter>,
  );
  test('Header component render', () => {
    expect(header.length).toEqual(1);
  });

  test('Link tag in Header component', () => {
    expect(header.find('Link')).toHaveLength(1);
  });

  test('<h1> tag in Header component', () => {
    expect(header.find('h1')).toHaveLength(1);
  });

  test('Section prop data type', () => {
    const sectionProp = header.find('.section__title').text();
    expect(typeof sectionProp).toBe('string');
  });

  test('Header component UI snapshot', () => {
    const header = create(
      <BrowserRouter>
        <Header />
      </BrowserRouter>,
    );
    expect(header.toJSON()).toMatchSnapshot();
  });
});
