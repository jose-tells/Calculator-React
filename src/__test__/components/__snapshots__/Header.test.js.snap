// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Header component UI snapshot 1`] = `
<header
  className="section"
>
  <a
    href="/"
    onClick={[Function]}
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-chevron-left fa-w-10 section__icon"
      data-icon="chevron-left"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 320 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </a>
  <h1
    className="section__title"
  />
</header>
`;
